.github/workflows/perf.yml

```diff
@@ -0,0 +1,153 @@
+      commit_id: ${{ steps.get-testing-version.outputs.commit_id }}
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.INTEG_TEST_AWS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
+          aws-region: us-west-2
+      
+      - name: Download Candidate from the latest commit
+        if: github.event_name != 'repository_dispatch'
+        run: aws s3 cp "s3://aws-otel-collector-release-candidate/${{ github.sha }}.tar.gz" ./candidate.tar.gz
+      
+      - name: Download Candidate base on dispatch payload
+        if: github.event_name == 'repository_dispatch'
+        run: aws s3 cp "s3://aws-otel-collector-release-candidate/${{ github.event.client_payload.sha }}.tar.gz" ./candidate.tar.gz
+      
+      - name: uncompress the candidate package
+        run: tar -zxf ./candidate.tar.gz
+          
+      - name: get testing version
+        id: get-testing-version
+        run: |
+          testing_version=`cat build/packages/TESTING_VERSION`
+          echo "::set-output name=testing_version::$testing_version"
+          commit_id=`cat build/packages/GITHUB_SHA`
+          echo "::set-output name=commit_id::$commit_id"
+
+  get-perf-test-cases:
+    runs-on: ubuntu-latest
+    outputs:
+      perf_matrix: ${{ steps.get-test-cases.outputs.soaking_matrix }}
+    steps:
+      - uses: actions/checkout@v2
+        
+      - name: Setup Python
+        uses: actions/setup-python@v2.1.4
+      
+      - name: get all the perf test cases (all the soaking test cases are perf test cases)
+        id: get-test-cases
+        run: |
+          perf_matrix=$(python e2etest/get-testcases.py perf_matrix)
+          echo "::set-output name=perf_matrix::$perf_matrix"
+          
+  run-perf-test:
+    runs-on: ubuntu-latest
+    needs: [get-perf-test-cases, get-testing-version]
+    strategy:
+      matrix: ${{ fromJson(needs.get-perf-test-cases.outputs.perf_matrix) }}
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+        
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.INTEG_TEST_AWS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
+          aws-region: us-west-2
+          
+      - name: Set up JDK 1.11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.11
+      
+      - name: Set up terraform
+        uses: hashicorp/setup-terraform@v1
+      
+      - name: Check out testing framework
+        uses: actions/checkout@v2
+        with:
+          repository: 'aws-observability/aws-otel-collector-test-framework'
+          path: testing-framework
+          
+      - name: Run Soaking test
```

Comment:
Did you mean to name this "Soaking" or should it be "Performance"?

Reply:
thx for pointing out! i will modify it